### ISSUE TITLE
[BUG] Model checkpoints are not saved using .save() after using .historical_forecasts()

### ISSUE THREAD

**Issue by StephanAkkerman** [unit8co/darts#1825]
**Describe the bug**
When training and evaluating a model using .historical_forecasts(), the model's checkpoint is not saved when using model.save(). Only the .pt file is saved but no .ckpt file is created.

**To Reproduce**
Use to code below to reproduce.

```py
import pandas as pd

from darts.models import NBEATSModel
from darts.dataprocessing.transformers import Scaler
from darts.datasets import AirPassengersDataset
from darts.models import NBEATSModel
from darts.metrics import rmse

# Read data:
series = AirPassengersDataset().load()

# Create training and validation sets:
train, val = series.split_after(pd.Timestamp(year=1957, month=12, day=1))

# Normalize the time series (note: we avoid fitting the transformer on the validation set)
transformer = Scaler()
transformer.fit(train)
train = transformer.transform(train)
val = transformer.transform(val)


def save_model():
    model = NBEATSModel(
        input_chunk_length=24,
        output_chunk_length=1,
        n_epochs=1,
        save_checkpoints=False,
    )

    # Train the model
    model.historical_forecasts(
        series=series,
        start=len(train),
        forecast_horizon=1,
        stride=1,
        retrain=True,
        verbose=False,
    )

    model.save("model.pt")


def load_model():
    loaded = NBEATSModel.load(f"model.pt")

    # Train the model
    pred = loaded.historical_forecasts(
        series=series,
        start=len(train),
        forecast_horizon=1,
        stride=1,
        retrain=False,
        verbose=False,
    )

    # Calculate the metrics
    model_rmse = rmse(val, pred)
    print(model_rmse)


save_model()
load_model()
```

**Expected behavior**
If you set `save_checkpoints=False` numerous checkpoints get saved however, I just want one final checkpoint which I assumed .save() would be able to do.

**System (please complete the following information):**
 - Python version: 3.10.11
 - darts version 0.24.0

**Additional context**

**Comment by madtoinou** [comment]
Hi @StephanAkkerman,

`historical_forecasts()` does not modify the original model but relies on a copy, the lines responsible for this behavior can be found [here](https://github.com/unit8co/darts/blob/master/darts/models/forecasting/forecasting_model.py#L1050-L1051).

If you want to train your model, you should use `fit()` as `historical_forecasts()` is rather meant to be used for evaluation.

**Comment by StephanAkkerman** [comment]
@madtoinou Thanks for your quick response. Is using `historical_forecassts()` not suitable for training a model using one-step ahead forecasts?

**Comment by madtoinou** [comment]
Since `historical_forecasts()` repeatedly builds a training set: either expanding from the beginning of series or moving with a fixed length train_length [doc](https://unit8co.github.io/darts/generated_api/darts.models.forecasting.linear_regression_model.html?highlight=historical#darts.models.forecasting.linear_regression_model.LinearRegressionModel.historical_forecasts), it's not the ideal approach to obtain checkpoints for your model as it will be retrained several times with different training sets.

**Comment by StephanAkkerman** [comment]
@madtoinou Thanks, that's clear. I think I misunderstood the process of training a model for one-step-ahead forecasting, thinking I should retrain the model constantly whereas I should just train it once and evaluate it without retraining at every step. Would using `fit()` and then using `historical_forecasts(retrain=False)` be the best approach for one-step-ahead forecasting in Darts?

Thanks again for your help, I appreciate it

**Comment by madtoinou** [comment]
Correct, `historical_forecasts(retrain=True)` will allow you to obtain an even better estimation of "how would this model do if it was used it in the past" since you prevent data leakage and mimic a real-life scenario.

`retrain=False` is usually used for model that are slow to train, to estimate the "historical" performance while saving time, ideally on a validation set (so that the model has never seen the values to forecast and avoid bias).

Glad to be of help.
